runtime/lua/vim/lsp/buf.lua

```diff
@@ -46,6 +46,14 @@ local function request(method, params, handler)
   return vim.lsp.buf_request(0, method, params, handler)
 end
 
+local function pos_request(method, params, handler)
+  validate {
+    method = {method, 's'};
+    handler = {handler, 'f', true};
+  }
```

Comment:
Why add the validation here instead of within `buf_request_position` itself?

Reply:
Fixed.